### PR TITLE
Support/testing library lld

### PR DESCRIPTION
**Pull request by KVNLS**
### 📝 Description

Update jest configuration in LLD to run integration tests using react testing library.
This PR fix:
- ledgerhq libraries exports
- remove the need to use moduleNameMapper for internal libraries
- remove unused babel configuration for lld
- Add a custom render function as described in the testing library documentation

### ❓ Context

- **Impacted projects**: `lld`, `llm` 
- **Linked resource(s)**: `` 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

You must be able to run tests in lld without errors.

### CODE REVIEW

File: apps/ledger-live-desktop/jest.config.js [hunk 1]
```diff
@@ -21,8 +30,10 @@ const defaultConfig = {
   testEnvironment: "node",
   testPathIgnorePatterns,
   globalSetup: "<rootDir>/tests/setup.ts",
-  setupFiles: ["<rootDir>/tests/jestSetup.ts"],
   moduleDirectories: ["node_modules"],
+  modulePaths: [compilerOptions.baseUrl],
+  setupFiles: ["jest-canvas-mock", "<rootDir>/tests/jestSetup.ts"],
+  moduleNameMapper: pathsToModuleNameMapper(compilerOptions.paths),
```
Comment (valpinkman) [inline apps/ledger-live-desktop/jest.config.js:36]: nice

File: apps/ledger-live-desktop/jest.config.js [hunk 2]
```diff
@@ -36,23 +47,25 @@ module.exports = {
...
       testRegex: "(/__tests__/.*|(\\.|/)react\\.test|spec)\\.tsx",
       testPathIgnorePatterns,
       moduleNameMapper: {
-        // Force module uuid to resolve with the CJS entry point, because Jest does not support package.json.exports. See https://github.com/uuidjs/uuid/issues/451
-        uuid: require.resolve("uuid"),
+        ...defaultConfig.moduleNameMapper,
         "~/(.*)": "<rootDir>/src/$1",
         "\\.(jpg|ico|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$":
           "<rootDir>/fileMock.js",
-        "@ledgerhq/domain-service(.*)": [
```
Comment (valpinkman) [inline apps/ledger-live-desktop/jest.config.js:52]: 🤩

File: libs/ui/packages/react/package.json [hunk 3]
```diff
@@ -123,5 +123,13 @@
     "stylelint-processor-styled-components": "^1.10.0",
     "tiny-glob": "^0.2.9",
     "webpack": "*"
+  },
+  "dependenciesMeta": {
```
Comment (valpinkman) [inline libs/ui/packages/react/package.json:127]: GG

**Review by valpinkman — APPROVED** [review]
Nice setup, nice clean up (thank you for the babel removal 🙏 )

LGTM

**Review by gre — APPROVED** [review]

**Comment by KVNLS** [comment]
I broke something for playwright I need to investigate (maybe babel was needed for the test build :/ )
btw I found that in tools/utils/index.js
```
// This is not the best way to do this, but it works for now.
        // The problem is that vitejs has trouble resolving everything under the /bridge subfolder.
        // Even though the files are there, it can't find them - and it manages to resolve other paths just fine.
        "@ledgerhq/coin-framework": path.join(
          path.resolve(__dirname, "..", "..", "..", "..", "libs", "coin-framework"),
          "lib-es",
        ),
```
I think that this is fixed now with exports definition I added.

**Pull request merged**